File: add/codesnippet/VisualBasic/m-system.windows.forms.o_0_2.vb
    ' Gets the caret width based upon the operating system or default value.
    Private Function GetCaretWidth() As Integer

        ' Check to see if the operating system supports the caret width metric. 
        If OSFeature.IsPresent(SystemParameter.CaretWidthMetric) Then

            ' If the operating system supports this metric,
            ' return the value for the caret width metric. 

            Return SystemInformation.CaretWidth
        Else

            ' If the operating system does not support this metric,
            ' return a custom default value for the caret width.

            Return 1
        End If
    End Function

File: add/codesnippet/CSharp/m-system.windows.forms.o_0_2.cs
        // Gets the caret width based upon the operating system or default value.
        private int GetCaretWidth ()
        {    

            // Check to see if the operating system supports the caret width metric. 
            if (OSFeature.IsPresent(SystemParameter.CaretWidthMetric))
            {

                // If the operating system supports this metric,
                // return the value for the caret width metric. 

                return SystemInformation.CaretWidth;
            } else

                // If the operating system does not support this metric,
                // return a custom default value for the caret width.

                return 1;
        }